### PR TITLE
backend: dwarfdbginf: add DW_AT_artificial to main subprogram when generated

### PR DESCRIPTION
**Pull request by ljmf00**
Fixes 22471.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

**Comment by ljmf00** [comment]
I'm currently writing another patch to be possible to add tests, since the compiler segfaults on `-main -c` for some reason.

File: src/dmd/backend/cdef.d [hunk 1]
```diff
@@ -605,6 +605,7 @@ enum
     CFG2expand      = 0x4000,  // expanded output to list file
     CFG2stomp       = 0x8000,  // enable stack stomping code
     CFG2gms         = 0x10000, // optimize debug symbols for microsoft debuggers
+    CFG2genmain     = 0x20000, // main entrypoint is generated
```
Comment (ljmf00) [inline src/dmd/backend/cdef.d:608]: I don't know the meaning of each flag enum, so I added to what I saw more appropriate. If there is any meaning to it, please tell me to add proper documentation to it.

**Comment by ljmf00** [comment]
> I'm currently writing another patch to be possible to add tests, since the compiler segfaults on `-main -c` for some reason.

Fixed on https://github.com/dlang/dmd/pull/13302 .

**Review by thewilsonator — APPROVED** [review]
Is this good to go?

**Comment by ljmf00** [comment]
> Is this good to go?

From my side yes. I'm just not sure if that flag is in the right place.

**Review by RazvanN7 — APPROVED** [review]

**Comment by PetarKirov** [comment]
I've rebased this PR on master as it was stuck waiting for Semaphore for quite a while.

**Pull request merged**